File: src/main/java/dipleks/model/en/Favorites.java
package dipleks.model.en;

import dipleks.database.ListWordsDatabase;
import dipleks.database.entity.DictionaryEntity;
import javafx.collections.ObservableList;

public class Favorites {

    public static void getListFavorites(ObservableList<DictionaryEntity> listWord) {
        try {
            for (DictionaryEntity d : listWord) {
                boolean isSelected = d.getFavorites().isSelected();
                ListWordsDatabase.changeFavoritesList(isSelected, d.getOriginal());
            }
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/java/dipleks/model/en/Favorites.kt
package dipleks.model.en

import dipleks.database.ListWordsDatabase
import dipleks.database.entity.DictionaryEntity
import javafx.collections.ObservableList

class Favorites {

    fun getListFavorites(listWord : ObservableList<DictionaryEntity>) =
        try {
            for (dictionary in listWord) {
                val isSelected = dictionary.favorites.isSelected
                ListWordsDatabase.changeFavoritesList(isSelected, dictionary.original)
            }
        } catch (e : Exception) {
            throw RuntimeException(e)
        }
}